typescript/packages/jumble/src/views/Shell.tsx

```diff
@@ -25,25 +34,57 @@ setIframeContextHandler({
   },
 });
 
-export default function Shell() {
+function Content() {
   const [count, setCount] = useState(0);
 
   const incrementCount = () => {
     setCount((c) => c + 1);
   };
 
   return (
-    <div className="h-full relative">
+    <>
       <button onClick={incrementCount} className="mb-4 px-4 py-2 bg-blue-500 text-white rounded">
         Increment Count ({count})
       </button>
 
       <CharmRunner
         charmImport={() => import("@/recipes/smol.tsx")}
         argument={{ count }}
-        className="border border-red-500 mt-4 p-2"
+        className="w-full h-full"
         autoLoad
       />
+    </>
+  );
+}
+
+export default function Shell() {
+  const [sidebarTab] = useCell(sidebar);
+
+  return (
+    <div className="h-full relative">
+      <WebComponent
+        as={"os-chrome"}
+        wide={sidebarTab === "source" || sidebarTab === "data" || sidebarTab === "query"}
+        locationTitle="Hello World"
+        onLocation={() => {
+          debugger;
+        }}
+      >
+        <Content />
+
+        <WebComponent
+          slot="overlay"
+          as="os-fabgroup"
+          className="pin-br"
+          onSubmit={() => {
+            console.log("submitted");
+          }}
+        />
```

Comment:
we can define interfaces and thin components for commonly used webcomponents using this same abstraction

it's a little bit automagical in terms of handling event names but... it works!